multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/security/AuthorizationChecker.java

```diff
@@ -69,6 +70,8 @@ public void ensureUserIsAuthorized(HttpServletRequest request, UserInfo userInfo
                 String message = MessageFormat.format(Messages.UNAUTHORISED_OPERATION_SPACE_ID, action, spaceGuid);
                 failWithForbiddenStatus(message);
             }
+        } catch (IllegalArgumentException e) {
```

Comment:
This is not the right place for this `catch`. As it is now, it would catch **all** `IllegalArgumentException`s, whereas you only want to catch the one that may occur when parsing the space GUID. For example, it would also catch an exception coming from the following line:
```
UUID userGuid = UUID.fromString(userInfo.getId());
```
If this were to happen, the user would receive `404 Not Found`, whereas the appropriate status code would be either `400 Bad Request`, `401 Unauthorized` or `403 Forbidden`.
And this is just an example. In the future, we may add more code to the `checkPermissions` method, and it may also throw `IllegalArgumentException`s.

Furthermore, when reading this code, it's impossible to know why the `try` block would even throw such exceptions. You have to actually go inside the `checkPermissions` method to figure out what's going on, and this is never a good sign. When reading a method, a person should be able to figure out why things are the way they are, without having to dig into the lower levels of abstraction.